File: app/src/main/java/com/xmartlabs/template/model/Session.java
package com.xmartlabs.template.model;

import com.xmartlabs.base.core.model.SessionType;

import org.parceler.Parcel;

import lombok.Data;
import lombok.ToString;

@Data
@Parcel
@ToString(callSuper = true)
public class Session implements SessionType {
  String accessToken;
}

File: template/app/src/main/java/com/xmartlabs/template/model/Session.kt
package com.xmartlabs.template.model

import com.xmartlabs.bigbang.core.model.SessionType

import lombok.Data

class Session: SessionType {
  var token: String? = null
  
  override fun setAccessToken(accessToken: String?) {
    token = accessToken
  }
  
  override fun getAccessToken() = token
}